tests/test_backend.py

```diff
@@ -1117,6 +1122,37 @@ def test_actual_expiry(self):
 
 
 class TestDefaultClient(unittest.TestCase):
+    def setUp(self):
+        self.client = caches[DEFAULT_CACHE_ALIAS].client
+
+    def tearDown(self):
+        try:
+            self.client.clear()
+        except exceptions.ConnectionInterrupted:
+            pass
+        finally:
+            self.client = None
+
+    def test_close_client_disconnect_default(self):
+        self.client.get_client(write=True)
+        with patch("redis.connection.ConnectionPool.disconnect") as mock:
+            self.client.close()
+            assert mock.called
+
+    @override_settings(DJANGO_REDIS_CLOSE_CONNECTION=False)
+    def test_close_disconnect_settings(self):
+        self.client.get_client(write=True)
+        with patch("redis.connection.ConnectionPool.disconnect") as mock:
+            self.client.close()
+            assert not mock.called
+
+    def test_close_disconnect_client_options(self):
```

Comment:
Why not change the options, via `override_settings`? it is easier to read for the people that come after you

Reply:
done

Reply:
done

just for the sake of sharing: setting_changed signals is clearing the test fixture: 
needs to populate again the client connection(s).
```
    def test_close_disconnect_settings_cache(self):
        settings.CACHES[DEFAULT_CACHE_ALIAS]["OPTIONS"]["CLOSE_CONNECTION"] = True
        with override_settings(CACHES=settings.CACHES):
            # enabling override_settings context emits 'setting_changed' signal
            # (re-set the value to populate again client connections)
            self.cache.set("test_close_disconnect_settings_cache", 0)
```